src/1.18/chunk.js

```diff
@@ -28,12 +28,15 @@ module.exports = (ChunkColumn, registry) => {
 
         const biomePalette = biomeSection.data.palette ? biomeSection.data.palette : [biomeSection.data.value] // "SingleValue" palette mess
 
-        const bitsPerBlock = Math.ceil(Math.log2(blockPalette.length))
+        let bitsPerBlock = Math.ceil(Math.log2(blockPalette.length))
         const bitsPerBiome = Math.ceil(Math.log2(biomePalette.length))
 
         if (!bitsPerBlock) {
           blockStates = nbt.comp({ palette: nbt.list(nbt.comp(blockPalette)) })
         } else {
+          if (bitsPerBlock === 1 || bitsPerBlock === 2 || bitsPerBlock === 3) {
```

Comment:
the test was already there, so I added it back:
https://github.com/PrismarineJS/prismarine-provider-anvil/actions/runs/9491519760/job/26157101354?pr=79

Reply:
I don't understand. Are you saying tests in the current master branch are failing without this? I don't understand the reasoning behind this change, it doesn't make sense to me because this will definitely have an effect on decoding blocks.

Reply:
But it is in the writing part of the code, not reading, isn't it? And yes tests are failing without this. Reverting this change and tests change will not fix the issue itself , for me saving didn't work most of the time. And it was the simple way to fix the problem . If you have any other suggestions please let me know 🙏

Reply:
This is definitely does not seem correct

Reply:
yes this is definitely wrong except if we find a bug in mojang code that would explain it